helios.js

```diff
@@ -28652,6 +28724,48 @@ class InlineDatum extends Datum {
 	}
 }
 
+class TxMetadata {
+	/**@type {Object.<number, Object.<string, any> | string>} */
+	#metadata;
+
+	constructor() {
+		this.#metadata = {};
+	}
+
+	/**
+	 *
+	 * @param {number} tag
+	 * @param {Object.<string, any> | string} data
+	 */
+	add(tag, data) {
+		this.#metadata[tag] = data;
+	}
+
+	dump() {
+		return Object.keys(this.#metadata).map((key) => ({
+			[key]: typeof this.#metadata[key] === 'string' ? this.#metadata[key]: JSON.stringify(this.#metadata[key]),
+		}))
+	}
+
+	toCbor() {
+		const data = Object.keys(this.#metadata).map((key) => [
+			CborData.encodeInteger(BigInt(key)),
+			CborData.encodeUtf8(typeof this.#metadata[key] === 'string' ? this.#metadata[key]: JSON.stringify(this.#metadata[key]), true),
+		]);
+		// @ts-ignore - FIXME: I don't know how to make this happy yet!
+		return CborData.encodeMap(data);
+	}
+
+	/**
+	 * #param {number[]} data
+	 */
+	static fromCbor(data) {
+		const cborMap = CborData.decodeMap(data, (pairBytes) => {
+			
```

Comment:
some example inner code to get you started:
```
let key = Number(CborData.decodeInteger(pairBytes));
let str = CborData.decodeUtf8(pairByes);
data[key] = str;
```

Of course the decode logic would be more complex because different types are possible (see line 7447 for an example)

Reply:
Thank you!!!